src/jdk.jpackage/share/classes/jdk/jpackage/internal/Platform.java

```diff
@@ -115,6 +121,10 @@ static boolean isLinux() {
         return getPlatform() == LINUX;
     }
 
+    static boolean is64Bit() {
+        return is64b;
+    }
+
```

Comment:
This is Windows specific thing. There is no point to keep it in shared code. I'd suggest to move this function in WixSourcesBuilder class.

I think it is not correct to check for arch of the OS where jpackage runs. Jpackage from 32bit JDK (if somebody would build 32bit OpenJDK) would would create msi installer on 64bit Windows it will still put 32bit Java runtime in app image. So the arch of installer is determined by arch of Java runtime baked in the app image. For simplicity we can assume that arch of Java runtime is the same as the arch of jpackage (this might not be the case for external Java runtime case though, but let's leave this possibility aside). If so `sun.arch.data.model` can be used instead of `os.arch`.

Reply:
Sorry, I left this comment earlier, but it was lost.

Reply:
I agree is better to have the check in windows specific code (not Platform) and will move it.
The "os.arch" System property in 32  java is "x86" regardless of whether running on 64 or 32 bit version of Windows, It reflects the arch of the running JVM (not the arch of the os as the name implies), so checking if system property "os.arch" is "x86" is the same as checking if "sun,arch.data.model" is "32".  Since "os.arch" is the more documented property I think we should continue to use that.

Reply:
(note:) I do build 32 bit java on windows to test this, I can "make CONF=windows-x86 images", but building installers is not supported and currently building test has problems in hotspot.